src/Features/CSharp/Portable/UsePatternMatching/CSharpIsAndCastCheckCodeFixProvider.cs

```diff
@@ -0,0 +1,110 @@
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UsePatternMatching
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal partial class CSharpIsAndCastCheckCodeFixProvider : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.InlineIsTypeCheckId);
+
+        public override FixAllProvider GetFixAllProvider() => new InlineTypeCheckFixAllProvider(this);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+                context.Diagnostics);
+            return SpecializedTasks.EmptyTask;
+        }
+
+        private Task<Document> FixAsync(
+            Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            return FixAllAsync(document, ImmutableArray.Create(diagnostic), cancellationToken);
+        }
+
+        private async Task<Document> FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var editor = new SyntaxEditor(root, document.Project.Solution.Workspace);
+
+            foreach (var diagnostic in diagnostics)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                AddEdits(root, editor, diagnostic, cancellationToken);
+            }
+
+            var newRoot = editor.GetChangedRoot();
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private void AddEdits(
+            SyntaxNode root,
+            SyntaxEditor editor,
+            Diagnostic diagnostic,
+            CancellationToken cancellationToken)
+        {
+            var ifStatementLocation = diagnostic.AdditionalLocations[0];
+            var localDeclarationLocation = diagnostic.AdditionalLocations[1];
+
+            var ifStatement = (IfStatementSyntax)ifStatementLocation.FindNode(cancellationToken);
+            var localDeclaration = (LocalDeclarationStatementSyntax)localDeclarationLocation.FindNode(cancellationToken);
+            var isExpression = (BinaryExpressionSyntax)ifStatement.Condition;
+
+            var updatedCondition = SyntaxFactory.IsPatternExpression(
+                isExpression.Left, SyntaxFactory.DeclarationPattern(
+                    ((TypeSyntax)isExpression.Right).WithoutTrivia(),
+                    Extensions.SyntaxTokenExtensions.WithoutTrivia(
+                        localDeclaration.Declaration.Variables[0].Identifier)));
```

Comment:
same reason as before.  There's an ambiguity between extension methods an the compiler can't resolve it.  I want to go in to see if i can fix things up here in hte future.